base/path.jl

```diff
@@ -246,8 +246,70 @@ function splitpath(p::String)
     return out
 end
 
+joinpath(path::AbstractString) = path
+
+if Sys.iswindows()
+
+function joinpath(path::AbstractString, paths::AbstractString...)
+    result_drive, result_path = splitdrive(path)
+
+    local p_drive, p_path
+    for p in paths
+        p_drive, p_path = splitdrive(p)
+
+        if startswith(p_path, ('\\', '/'))
+            # second path is absolute
+            if !isempty(p_drive) || !isempty(result_drive)
+                result_drive = p_drive
+            end
+            result_path = p_path
+            continue
+        elseif !isempty(p_drive) && p_drive != result_drive
+            if lowercase(p_drive) != lowercase(result_drive)
+                # different drives, ignore the first path entirely
+                result_drive = p_drive
+                result_path = p_path
+                continue
+            end
+            # same drive in different case
+            result_drive = p_drive
+        end
+
+        # second path is relative to the first
+        if !isempty(result_path) && result_path[end] ∉ ('\\', '/')
```

Comment:
Should I use  `last` and `first` instead of `x[end]` and `x[1]`. I'm not sure if there is a convention or advantage to one or the other when `x` is a string?

Reply:
They should be equivalent so whichever seems clearer to you works.

Reply:
I think what you're doing now is clearer and how I would write it.